libs/features/mas/src/hydrate.js

```diff
@@ -67,12 +68,18 @@ export function processMnemonics(fields, merchCard, mnemonicsConfig) {
 }
 
 function processBadge(fields, merchCard, mapping) {
-    if (fields.variant === 'plans' || fields.variant === 'plans-students') {
+    if (fields.variant === 'plans' || fields.variant === 'plans-students' || fields.variant === 'fries') {
```

Comment:
I agree with @npeltier  we should add `badge:true` to the variant mapping and avoid fragmenting variant mapping logic.

Reply:
I have changed `hydrate.js` to process fields based on mapping slot availability of variants.